code/lib/cli/src/automigrate/fixes/new-frameworks/new-frameworks.ts

```diff
@@ -0,0 +1,528 @@
+      return null;
+    }
+
+    const configDir = userDefinedConfigDir || configDirFromScript;
+    let mainConfig;
+    try {
+      mainConfig = await loadMainConfig({ configDir });
+    } catch (err) {
+      throw new Error(
+        dedent`Unable to find or evaluate ${chalk.blue(mainConfigPath)}: ${err.message}`
+      );
+    }
+
+    let frameworkPackage =
+      typeof mainConfig.framework === 'string' ? mainConfig.framework : mainConfig.framework?.name;
+    const hasFrameworkInMainConfig = !!frameworkPackage;
+
+    if (!hasFrameworkInMainConfig) {
+      frameworkPackage = await detectFramework(packageJson);
+      // This scenario will only happen when user's have things properly set up for SB 7
+      // but do not contain a framework field in main.js
+      if (frameworkPackage) {
+        mainConfig.framework = frameworkPackage;
+      }
+    }
+
+    // if --renderer is passed to the command, just use it.
+    // Useful for monorepo projects to automate the script without getting prompts
+    let rendererPackage = userDefinedRendererPackage;
+    if (!rendererPackage) {
+      // at some point in 6.4 we introduced a framework field, but filled with a renderer package
+      if (frameworkPackage && Object.keys(rendererPackages).includes(frameworkPackage)) {
+        rendererPackage = frameworkPackage;
+      } else {
+        // detect the renderer package from the user's dependencies, and if multiple are there (monorepo), prompt the user to choose
+        rendererPackage = await detectRenderer(packageJson);
+      }
+    }
+
+    const builderConfig = mainConfig.core?.builder;
+
+    // bail if we can't detect an official renderer
+    const supportedPackages = Object.keys(packagesMap);
+    if (!supportedPackages.includes(rendererPackage)) {
+      return null;
+    }
+
+    const builderInfo = getBuilderInfo(mainConfig);
+
+    let newFrameworkPackage = packagesMap[rendererPackage][builderInfo.name];
+
+    // bail if there is no framework that matches the renderer + builder
+    if (!newFrameworkPackage) {
+      return null;
+    }
+
+    const renderer = rendererPackages[rendererPackage];
+    // @ts-expect-error account for renderer options for packages that supported it: reactOptions, angularOptions. (svelteOptions got removed)
+    let rendererOptions = mainConfig[`${renderer}Options`] || {};
+
+    const frameworkOptions =
+      typeof mainConfig.framework === 'string' ? {} : mainConfig.framework?.options;
+
+    let dependenciesToRemove = [
+      '@storybook/builder-webpack5',
+      '@storybook/manager-webpack5',
+      '@storybook/builder-webpack4',
+      '@storybook/manager-webpack4',
+      '@storybook/builder-vite',
+      'storybook-builder-vite',
+    ];
+
+    let dependenciesToAdd: string[] = [];
+    let addonsToRemove: string[] = [];
+    let addonOptions = {};
+
+    // Next.js specific automigrations
+    if (allDependencies.next && semver.gte(semver.coerce(allDependencies.next).version, '12.0.0')) {
```

Comment:
If you want to go the extra mile, you could warn Next.js users below 12.2.0 that not the whole feature set is supported (e.g. `next/future/image`)

Reply:
I think that's a little too much hehe